### PR TITLE
Lint tests

### PR DESCRIPTION
**Pull request by jaller94**
Lint our tests as well.

### CODE REVIEW

**Review by Half-Shot — APPROVED** [review]
No major nits from me other than the two. Thanks!

File: .eslintrc [hunk 1]
```diff
@@ -93,5 +93,23 @@
...
+    {
+      "files": [
+        "spec/**/*.js"
+      ],
+      "rules": {
+        "@typescript-eslint/no-this-alias": "off", // I'm unsure if we should disallow this.
+        "@typescript-eslint/no-empty-function": "off",
+        "@typescript-eslint/no-var-requires": "off",
+        "strict": "off",
+        // temporary
```
[the remark this answers is not in the document]
Comment (jaller94) [reply .eslintrc:107]: These had many errors that would make the PR even more noisy and more likely to have serious merge conflicts.

File: .eslintrc [hunk 2]
```diff
@@ -93,5 +93,23 @@
...
     ]
-  }
+  },
+  "overrides": [
+    {
+      "files": [
+        "spec/**/*.js"
+      ],
+      "rules": {
+        "@typescript-eslint/no-this-alias": "off", // I'm unsure if we should disallow this.
```
[the remark this answers is not in the document]
Comment (jaller94) [reply .eslintrc:103]: This is about `var self = this;`

e.g. https://github.com/matrix-org/matrix-appservice-irc/blob/948f3cc127eb6e85bd5c3436db24c93517235d45/spec/util/irc-client-mock.js#L54

[the remark this answers is not in the document]
Comment (Half-Shot) [reply .eslintrc:103]: Fair, old code :(

File: .eslintrc (hunk 1 above)
[the remark this answers is not in the document]
Comment (Half-Shot) [reply .eslintrc:107]: Ah, I see.

**Pull request merged**